apis/refs/v1beta1/kmskeyref.go

```diff
@@ -14,11 +14,97 @@
 
 package v1beta1
 
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
 type KMSCryptoKeyRef struct {
-	/* The KMSCryptoKeyRef selfLink, when not managed by KCC. */
+	//  If provided must be in the format `projects/[kms_project_id]/locations/[region]/keyRings/[key_ring_id]/cryptoKeys/[key]`.
```

Comment:
I've been thinking about this, I feel checks on External is unnecessary. KCC does not validate External format for now, and for example ComputeAddressRef, its External field accepts ID, full URL, or an address. I do not think KCC has a fixed pattern for External field, and should let GCP to fail the request if the value if invalid. I'll have a PR to discuss this, before that let's process with the current approach.